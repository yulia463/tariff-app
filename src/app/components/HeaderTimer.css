.header-timer {
    width: 100%;
    background-color: #1D5B43;
    border-top-left-radius: 60px;
    border-top-right-radius: 60px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 16px 0;
}

.header-text {
    font-weight: 600;
    font-size: 24px;
    line-height: 130%;
    text-align: center;
    color: white;
    margin-bottom: 8px;
}

.timer-wrapper {
    display: flex;
    align-items: center;
    gap: 8px;
    font-weight: 700;
    font-size: 32px;
    line-height: 110%;
}

.star {
    font-size: 24px;
}

/* Адаптив */
@media (max-width: 375px) {
    .header-timer {
        border-radius: 0;
    }

    .header-text {
        font-size: 18px;
    }

    .timer-wrapper {
        font-size: 32px;
    }
}

@media (max-width: 320px) {
    .header-text {
        font-size: 14px;
        text-transform: uppercase;
    }

    .timer-wrapper {
        font-size: 28px;
    }

    .star {
        font-size: 20px;
    }
}
