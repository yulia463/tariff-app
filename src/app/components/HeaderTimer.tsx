'use client'
import React from 'react'
import { useCountdown } from '@/app/hooks/useCountdown'
import './HeaderTimer.css'

export default function HeaderTimer({
                                      startSeconds = 960, // 16 минут
                                      onExpire,
                                    }: {
  startSeconds?: number
  onExpire?: () => void
}) {
  const { secondsLeft, minutes, seconds } = useCountdown(startSeconds)

  React.useEffect(() => {
    if (secondsLeft === 0 && onExpire) onExpire()
  }, [secondsLeft, onExpire])

  // Цвет таймера
  let timerColor = '#FFBB00'
  if (secondsLeft <= 2 * 60 + 59 && secondsLeft > 0) timerColor = '#FF4E4E'
  if (secondsLeft === 0) timerColor = '#FFFFFF'

  return (
      <header className="header-timer">
        <div className="header-text">
          Успейте открыть пробную неделю
        </div>

        <div className="timer-wrapper" style={{ color: timerColor }}>
          <span className="star">✦</span>
          <span className="timer-number">{String(minutes).padStart(2, '0')}</span>
          <span className="timer-number">:</span>
          <span className="timer-number">{String(seconds).padStart(2, '0')}</span>
          <span className="star">✦</span>
        </div>
      </header>
  )
}
