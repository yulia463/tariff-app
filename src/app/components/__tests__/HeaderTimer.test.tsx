import React from "react";
import { render, screen } from "@testing-library/react";
import HeaderTimer from "../HeaderTimer";

describe("HeaderTimer", () => {
    test("renders with default values", () => {
        render(<HeaderTimer />);
        expect(screen.getByText("Успейте открыть пробную неделю")).toBeInTheDocument();
    });

    test("changes timer color when time is low", () => {
        render(<HeaderTimer startSeconds={120} />);
        const timer = screen.getByText(/:/);
        expect(timer).toBeInTheDocument();
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<HeaderTimer startSeconds={960} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
