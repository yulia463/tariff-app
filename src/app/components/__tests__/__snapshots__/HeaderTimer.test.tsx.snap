// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HeaderTimer matches snapshot 1`] = `
<DocumentFragment>
  <header
    class=" w-full bg-[#1D5B43] rounded-t-[60px] flex flex-col items-center justify-center py-4 max-[430px]:rounded-none "
  >
    <div
      class=" font-semibold text-white text-center mb-2 text-[24px] leading-[130%] max-[375px]:text-[18px] max-[320px]:text-[14px] max-[320px]:uppercase "
    >
      Успейте открыть пробную неделю
    </div>
    <div
      class="
           flex items-center
           font-bold 
           text-[32px]    
           leading-[110%] 
           max-[320px]:text-[28px]
    "
      style="color: rgb(255, 187, 0);"
    >
      <div
        class="text-[14px] pr-[8px]"
      >
        ✦
      </div>
      <span>
        16
      </span>
      <span>
        :
      </span>
      <span>
        00
      </span>
      <div
        class="text-[14px] pl-[8px]"
      >
        ✦
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`HeaderTimer совпадает со snapshot 1`] = `
<DocumentFragment>
  <header
    class=" w-full bg-[#1D5B43] rounded-t-[60px] flex flex-col items-center justify-center py-4 max-[430px]:rounded-none "
  >
    <div
      class=" font-semibold text-white text-center mb-2 text-[24px] leading-[130%] max-[375px]:text-[18px] max-[320px]:text-[14px] max-[320px]:uppercase "
    >
      Успейте открыть пробную неделю
    </div>
    <div
      class="
      flex items-center
      font-bold 
      text-[32px]    
      leading-[110%] 
      max-[320px]:text-[28px]
    "
      style="color: rgb(255, 187, 0);"
    >
      <div
        class="text-[14px] pr-[8px]"
      >
        ✦
      </div>
      <span>
        16
      </span>
      <span>
        :
      </span>
      <span>
        00
      </span>
      <div
        class="text-[14px] pl-[8px]"
      >
        ✦
      </div>
    </div>
  </header>
</DocumentFragment>
`;
